react/src/components/comparison-panel.tsx

```diff
@@ -452,20 +453,7 @@ function insert_missing(rows: ArticleRow[][], idxs: number[][]): ArticleRow[][]
 // eslint-disable-next-line prefer-function-component/prefer-function-component -- TODO: Maps don't support function components yet.
 class ComparisonMap extends MapGeneric<MapGenericProps & { longnames: string[], colors: string[] }> {
     override buttons(): ReactNode {
-        return (
-            <div style={{
-                display: 'flex', backgroundColor: '#fff8f0', padding: '0.5em', borderRadius: '0.5em',
-                alignItems: 'center',
-            }}
-            >
-                <span className="serif" style={{ fontSize: '15px', fontWeight: 500 }}>Zoom to:</span>
-                <div style={{ width: '0.25em' }} />
-                {this.zoom_button(-1, 'black', () => { this.zoom_to_all() })}
-                {this.props.longnames.map((longname, i) => {
-                    return this.zoom_button(i, this.props.colors[i], () => { this.zoom_to(longname) })
-                })}
-            </div>
-        )
+        return <ComparisonMapButtons map={this} />
```

Comment:
This is a huge sin.

Reply:
Wait why?

Reply:
The whole structure of this is fucked because we're using class components.

We should refactor the Map to be an functional component, but until then stuff like this is going to keep popping up.

The problem with class components is that dependency management is much harder compared to functional components.

Reply:
So no changes needed now, but this does me sad.